woocommerce/woocommerce
# Try: building blocks from a npm package

Closes https://github.com/woocommerce/woocommerce-gutenberg-products-block/issues/416, see https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/456

This PR adds the [@woocommerce/block-library](https://www.npmjs.com/package/@woocommerce/block-library) package as an dependency to package.json, and updates the grunt build process to pull in the compiled files from the npm package – the CSS and JS files, along with 2 PHP files. One php file, `class-wc-block-library.php` registers the blocks, and `class-wc-block-featured-product.php` handles rendering the Featured Product block.

### How to test the changes in this Pull Request:

There are two parts to this PR; the part where the blocks have been added to the repo, and the part where we actually build the blocks from the block-library package.

To test the first part:

1. Deactivate the blocks plugin if you have it
1. Load up the block editor on a post or a page
2. Add a new block by clicking the +
3. You should see a new section, "WooCommerce", with 8 blocks, including "Products by Category", "Hand-picked Products", "Newest Products", etc.
4. Adding one of these blocks works as expected
5. Publish & view the post
6. The post on the frontend looks the same as the preview

To test the build process:

1. Run `npm install`
2. Run `grunt blocks` or just `grunt` ¹
3. You'll see 3 "copy" tasks run, each copying over some files:

```
Running "copy:js" (copy) task
Copied 10 files

Running "copy:css" (copy) task
Copied 3 files

Running "copy:php" (copy) task
Copied 2 files
```

After copying the files, you can run through the steps in the first part to test that the blocks are working correctly.

¹ I've noticed the grunt task doesn't complete without errors, so I'm not sure if I need to fix anything- the issues i see are in jshint & postcss.
